doc/markdownlinkconverter/markdownlinkconverter.c

```diff
@@ -0,0 +1,452 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <stdbool.h>
+#include <string.h>
+
+#if defined(WIN32)
+#define FOLDER_DELIMITER '\\'
+#else
+#define FOLDER_DELIMITER '/'
+#endif
+
+/* Slashes or backslashes to the elektra root.
+ * Depends on the place of the executable, example:
+ * /foo/bar/elektra/build/bin/executable
+ * E:\\foo\bar\elektra\build\bin\executable
+ * In both cases the there are 3 slashes or backslashes
+ */
+#define FOLDER_DELIMITER_TO_ELEKTRA_ROOT 3
+#define TEMP_FILENAME "temp"
+
+// Link Blacklist: do not convert links with the following starting and ending
+char * ignoreTargetEnd [] = { ".h", ".c" , ".cpp" , ".hpp" , "\0"};
+// ignore http, @ref and #anchor
+char * ignoreTargetStart [] = { "#", "@" , "http" , "\0"};
+// both need to be null terminated
+
+// helpers
+void printTarget(FILE * output, char * target, char * filenameInElektra);
+void printConvertedPath (FILE * output ,char * path);
+char * getPathInElektraRoot (char * executablename, char * filename);
+void exitError (FILE * f1, FILE * f2, char * mes);
+
+/* These structs represent the transitions from the state machines.
+ * Means if char c is read and the state machine is in state s
+ * the next state is t[c][s].
+ */
+struct transitionLink { int t[10][7]; };
+struct transitionTitle { int t[10][4]; };
+
+/* This transition array represents the regex for a link, namely:
+ *
+ * not(\!)\[.*\][' '\n\t]*\(.*\)
+ *
+ * 0 is the initial state and 5 the goal state.
+ * If the trap will be reached the input will be
+ * set back. This transition array matches the
+ * regex multiple times, witch is correct due to the
+ * fact that links can not be nested.
+ */
+
+#define LINK_START 0
+#define LINK_POSSIBLE(old, new) ((old == 0 || old == 5) && new == 2)
+#define LINK_FOUND(old, new) (state == 4 && newstate == 5)
+#define LINK_NAME(old, new) (((old == 2 || old == 3) && (new == 2  || new == 3)) \
+								|| (old == 3 && new == 4))
+#define LINK_TARGET(old, new) (state == 4 && newstate == 4)
+#define LINK_TRAP(old, new) (new == 6)
+#define LINK_NOLINK(old, new) (newstate == 0 || newstate == 1)
```

Comment:
I think you can should use inline functions here.
These macros have a number of problems. Think of `LINK_POSSIBLE(a++, b)` or even worse: `LINK_POSSIBLE(a&b, c)`.